app/main/views/frameworks.py

```diff
@@ -1123,10 +1005,10 @@ def contract_review(framework_slug, agreement_id):
             agreement_id, current_user.email_address, {'uploaderUserId': current_user.id}
         )
 
-        email_client = DMNotifyClient()
+        notify_client = DMNotifyClient()
         for email_address in returned_agreement_email_recipients(supplier_framework):
             try:
-                email_client.send_email(
+                notify_client.send_email(
                     to_email_address=email_address,
                     template_name_or_id="framework_agreement_signature_page",
                     personalisation={
```

Comment:
We should add in a `reference` kwarg here for the Notify client, with a hash of the email address (like we do for the clarification question email above)

Reply:
Wait a second - what email is this again? Is this `contract_review` view ever used for G8 and up?

Reply:
In the above, we use the current user's email address. Why is that? And is that what I should replicate?

Reply:
It's so that we can find the unique reference to it in the Notify dashboard if we need to. So just using the email address we're sending to in `to_email_address` is fine.

Reply:
> Wait a second - what email is this again? Is this `contract_review` view ever used for G8 and up?

yes - we removed the other route, but this is the one that's used for G8 and up

Reply:
Ah yes, sorry. This file is too confusing! So this is the one that's already using Notify, got it. We should still add a reference I think...